spot_rl_experiments/spot_rl/utils/geometry_utils.py

```diff
@@ -134,3 +137,117 @@ def is_pose_within_bounds(
     )
 
     return bool(is_within_linear_bounds and is_within_angular_bounds)
+
+
+def is_position_within_bounds(
+    test_position, target_position, xy_dist_threshold, z_dist_threshold
+) -> bool:
+    """
+    Check if a test position(x,y,z) is within  of target pose.
+
+    Args:
+        test_position (list): Test position, as [x,y,z]
+        target_position (list): Target position, as [x,y,z]
+        xy_dist_threshold (float): Threshold for xy distance (in meters)
+        z_dist_threshold (float): Threshold for z distance (in meters)
+
+    Returns:
+        is_within (bool): True if the test pose is within both linear and angular bounds of the target pose, False otherwise.
+    """
+    # Linear xy dist bounds
+    is_within_linear_xy_bounds = (
+        euclidean(test_position[:2], target_position[:2]) < xy_dist_threshold
+    )
+
+    # Linear z dist bounds
+    is_within_linear_z_bounds = (
+        abs(test_position[2] - target_position[2]) < z_dist_threshold
+    )
+
+    return bool(is_within_linear_xy_bounds and is_within_linear_z_bounds)
+
+
+def interpolation_between_two_positions(point1, point2, steps, kind="linear"):
+    """
+    Interpolate between two 3D points
```

Comment:
Linear interpolation

Reply:
The method, takes in the "kind" of interpolation as input.

It could be linear, even other types (like cubic, etc).